src/Evolution/Systems/Cce/Actions/BoundaryComputeAndSendToEvolution.hpp

```diff
@@ -123,6 +123,69 @@ struct BoundaryComputeAndSendToEvolution<H5WorldtubeBoundary<Metavariables>,
   }
 };
 
+/*!
+ * \ingroup ActionsGroup
+ * \brief Calculates the analytic boundary data at the specified `time`, and
+ * sends the resulting Bondi-Sachs boundary data to the `EvolutionComponent`
+ *
+ * \details This uses the `Cce::AnalyticBoundaryDataManager` to
+ * perform all of the work of calculating the analytic boundary solution, which
+ * in turn uses derived classes of `Cce::Solutions::WorldtubeData` to calculate
+ * the metric data before it is transformed to Bondi-Sachs variables.
+ *
+ * \ref DataBoxGroup changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies:
+ *   - `Tags::AnalyticWordltubeBoundaryDataManager`
+ */
+template <typename Metavariables, typename EvolutionComponent>
+struct BoundaryComputeAndSendToEvolution<
+    AnalyticWorldtubeBoundary<Metavariables>, EvolutionComponent> {
+  template <typename ParallelComponent, typename DbTagList, typename ArrayIndex>
+  static void apply(db::DataBox<DbTagList>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const TimeStepId& time) noexcept {
+    if constexpr (tmpl::list_contains_v<
+                      DbTagList,
+                      ::Tags::Variables<typename Metavariables::
+                                            cce_boundary_communication_tags>>) {
+      bool successfully_populated = false;
+      db::mutate<Tags::AnalyticBoundaryDataManager,
+                 ::Tags::Variables<
+                     typename Metavariables::cce_boundary_communication_tags>>(
+          make_not_null(&box),
+          [&successfully_populated, &time](
+              const gsl::not_null<Cce::AnalyticBoundaryDataManager*>
+                  worldtube_data_manager,
+              const gsl::not_null<Variables<
+                  typename Metavariables::cce_boundary_communication_tags>*>
+                  boundary_variables) noexcept {
+            successfully_populated =
+                (*worldtube_data_manager)
+                    .populate_hypersurface_boundary_data(
+                        boundary_variables, time.substep_time().value());
+          });
+
+      if (not successfully_populated) {
+        ERROR("Insufficient boundary data to proceed, exiting early at time " +
+              std::to_string(time.substep_time().value()));
```

Comment:
can't you just stream the value?  (i.e.   `+ std::to_string(...)` -> `<< time.substep_time().value`